lib/base/configuration.cpp

```diff
@@ -12,7 +30,95 @@ String Configuration::ApiBindHost;
 String Configuration::ApiBindPort{"5665"};
 bool Configuration::AttachDebugger{false};
 String Configuration::CacheDir;
-int Configuration::Concurrency{static_cast<int>(std::thread::hardware_concurrency())};
+
+#ifdef __linux__
+static std::string ReadSysLine(const char *file)
```

Comment:
This function looks overly complex for what it does. Shouldn't this basically be the same as `std::ifstream` + `std::getline`? The only thing I see that wouldn't be possible with that is the check for `ENOENT`. If this is just for kernel version compatibility, using an extra check would be fine. If this file can (dis)appear at runtime (changing the cgroup options, moving the process between cgroups), have a look at `boost::iostreams` like it's already used in `AtomicFile`.